tokio/src/net/tcp/socket.rs

```diff
@@ -480,7 +497,14 @@ impl TcpSocket {
     /// }
     /// ```
     pub fn listen(self, backlog: u32) -> io::Result<TcpListener> {
-        let mio = self.inner.listen(backlog)?;
+        self.inner.listen(backlog as i32)?;
+
+        let socket = std::mem::ManuallyDrop::new(self.inner);
+        #[cfg(windows)]
+        let mio = unsafe { mio::net::TcpListener::from_raw_socket(socket.as_raw_socket()) };
+        #[cfg(unix)]
+        let mio = unsafe { mio::net::TcpListener::from_raw_fd(socket.as_raw_fd()) };
```

Comment:
It seems like the `Socket` type in socket2 implements `IntoRawFd`, which seems more appropriate here than using `AsRawFd` with `ManuallyDrop`.

Reply:
done in https://github.com/tokio-rs/tokio/pull/4270/commits/216baeb220fb85bcfabb36b2ff20d44c0c3eb957